tests/integration/components/phone-input-test.js

```diff
@@ -39,6 +39,31 @@ module('Integration | Component | phone-input', function(hooks) {
     assert.dom('input').hasValue(newValue)
   })
 
+  test('renders the value with separate dial code option', async function(assert) {
+    assert.expect(3)
+
+    await this.owner.lookup('service:phone-input').load()
+
+    const newValue = '2'
+    this.set('separateDialNumber', null)
+    this.set('update', () => {})
```

Comment:
Do you mind simplifying the test a bit? 

Here you could set the update function to:
```
    this.set('update', value => {
      this.set('separateDialNumber', value)
    })
```
to not change it afterwards. I feel like modyfing it in the test scope makes things harder to understand.